<div class="alert alert-success">
<font size="5"><b>Комментарий ревьюера 2</b></font>

Привет. Теперь все сделано отлично. Метод сводных таблиц освоен. Он нам очень пригодится в дальнейших проектах. Ты проделала классную работу, молодец. Поздравляю с ее завершением. Надеюсь, проект был интересен и поставлен. Продолжай в том же духе. Успехов в дальнейшем пути :)
</div>

<div class="alert alert-info">
<font size="5"><b>Комментарий ревьюера</b></font>

Привет, Дана! Спасибо, что прислала задание:) Меня зовут Слепцов Артем и я буду проверять твой проект) Ты проделала большую работу над проектом. Он выполнен уже на достойном уровне. Однако есть моменты, которые еще можно улучшить. Будет здорово, если ты, надеюсь, не против, если я буду на ты, будешь отвечать на комментарии и участвовать в диалоге. Если обращение на ты неприемлемо, то прошу сообщить. 

Мои комментарии обозначены пометкой **Комментарий ревьюера**. Далее в файле ты сможешь найти их в похожих ячейках:
    
<div class="alert alert-success">Успех: Если фон комментария зелёный - всё сделано правильно. Рекомендации укажу таким же цветом;</div>
        
<div class="alert alert-warning">Совет: Оранжевый - некритичные замечания;</div>
        
<div class="alert alert-danger">Ошибка: Красный - нужно переделать. </div>
        
Не удаляй эти комментарии и постарайся учесть их в ходе выполнения данного проекта. Свои же комментарии ты можешь обозначать любым заметным способом. 

</div>

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
children_debt = data.groupby('children')['debt'].agg(['count', 'sum'])
children_debt['repayment_rate'] = (1 - children_debt['sum'] / children_debt['count']) * 100
display(children_debt)
# Ваш код будет здесь. Вы можете создавать новые ячейки.

count   sum  repayment_rate
children                             
0         14091  1063       92.456178
1          4808   444       90.765391
2          2052   194       90.545809
3           330    27       91.818182
4            41     4       90.243902
5             9     0      100.000000

<div class="alert alert-warning">
<font size="5"><b>Комментарий ревьюера</b></font>

Совет: Таблицы на печать выводи с помощью метода display. print переводит данные в строку, а нам этого не надо. 

</div>

**Вывод:** 
Клиенты без детей (0 детей) имеют самый высокий процент возврата в срок (92.46%).
Клиенты с 1 и 2 детьми показывают схожий процент возврата, чуть ниже (90.77% и 90.55% соответственно).
Клиенты с 3 детьми имеют немного лучший процент возврата (91.82%).
Клиенты с 4 детьми показывают наименьший процент возврата (90.24%), но разница незначительна.
Клиенты с 5 детьми показали 100% возврат, но это связано с малым количеством кредитов (всего 9).
Процент возврата в срок не сильно меняется с увеличением количества детей, и все значения выше 90%. Тем не менее, клиенты с 0 и 3 детьми показывают наилучшие результаты.
Малое количество данных: У клиентов с 4 и 5 детьми количество кредитов очень маленькое, что может не позволить сделать окончательные выводы по этим группам

<div class="alert alert-success">
<font size="5"><b>Комментарий ревьюера</b></font>

Успех: Вывод не противоречит полученному результату. 

</div>

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [23]:

family_debt = data.groupby('family_status')['debt'].agg(['count', 'sum', 'mean'])
family_debt['repayment_rate'] = (1 - family_debt['sum'] / family_debt['count']) * 100 
family_debt['debt_ratio'] = family_debt['mean'] * 100  
display(family_debt)

count  sum      mean  repayment_rate  debt_ratio
family_status                                                          
Не женат / не замужем   2796  273  0.097639       90.236052    9.763948
в разводе               1189   84  0.070648       92.935240    7.064760
вдовец / вдова           951   63  0.066246       93.375394    6.624606
гражданский брак        4134  385  0.093130       90.686986    9.313014
женат / замужем        12261  927  0.075606       92.439442    7.560558

<div class="alert alert-success">
<font size="5"><b>Комментарий ревьюера</b></font>
  
Совет: Круто, что ты знаешь метод agg. В него также можно добавить и метод mean для нахождения доли должников в каждой из групп. 

</div>

**Вывод:** 
Клиенты в разводе показывают процент возврата в срок 92.94%, что говорит о достаточно высоком уровне возвратов.
Вдовцы/вдовы имеют наивысший процент возврата (93.38%).
Клиенты, не состоящие в браке, показывают самый низкий процент возврата (90.24%).
В общем, данные показывают, что семейное положение может оказывать влияние на возврат кредита, но разница в процентах не является критической.

<div class="alert alert-success">
<font size="5"><b>Комментарий ревьюера</b></font>

Успех: Подумай также и над причинами получения такого результата. 

</div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [26]:

data['total_income'] = data['total_income'].apply(categorize_income)
income_debt = data.groupby('total_income')['debt'].agg(['count', 'sum'])
income_debt['repayment_rate'] = (1 - income_debt['sum'] / income_debt['count']) * 100
display(income_debt) # Ваш код будет здесь. Вы можете создавать новые ячейки.

count   sum  repayment_rate
total_income                             
A                25     2       92.000000
B              5014   354       92.939769
C             15921  1353       91.501790
D               349    21       93.982808
E                22     2       90.909091

<div class="alert alert-warning">
<font size="5"><b>Комментарий ревьюера</b></font>

Совет: Не совсем понимаю, зачем повторно проводиьт категоризацию. Мы же ее осуществили ранее, здесь делать ее повторно не нужно. 

</div>

**Вывод:** 
Категория A (самый высокий доход):

Уровень дохода A имеет 25 кредитов, из которых 2 не были возвращены, что соответствует проценту возврата 92.00%. Это высокий процент, но выборка мала, что может влиять на надежность результатов.
Категория B:

Наиболее крупная группа с 5014 кредитами и процентом возврата 92.94%. Это подтверждает, что клиенты со средним и высоким доходом имеют высокий уровень надежности.
Категория C:

Здесь наблюдается 15921 кредит, и процент возврата составляет 91.50%. Несмотря на большое количество кредитов, этот процент ниже, чем в категориях B и D, что может указывать на некоторые риски.
Категория D:

Процент возврата в срок 93.98% — самый высокий среди всех категорий. Это может указывать на то, что клиенты с доходом в этом диапазоне более надежны.
Категория E (самый низкий доход):

Имеет самый низкий уровень с 22 кредитами и процентом возврата 90.91%. Это также может указывать на риск невозврата среди клиентов с низким доходом.

В целом, клиенты с более высоким доходом имеют более высокий процент возврата кредита. Однако стоит отметить, что категория A имеет малое количество кредитов, что делает результаты менее надежными.
Низкий доход: Клиенты с низким доходом (категория E) показывают более низкий процент возврата, что может указывать на потенциальные риски для кредиторов.

<div class="alert alert-success">
<font size="5"><b>Комментарий ревьюера</b></font>

Успех: Соглашусь, не все группы имеют достаточное число клиентов для построения по ним надежных выводов. 

</div>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [31]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

purpose_debt_pivot = data.pivot_table(
    index='purpose_category',               
    values='debt',                         
    aggfunc=['count', 'sum', 'mean']       
)

purpose_debt_pivot.columns = ['total_loans', 'total_debts', 'debt_ratio']


purpose_debt_pivot['repayment_rate'] = (1 - purpose_debt_pivot['total_debts'] / purpose_debt_pivot['total_loans']) * 100

display(purpose_debt_pivot) # Ваш код будет здесь. Вы можете создавать новые ячейки.

total_loans  total_debts  debt_ratio  repayment_rate
purpose_category                                                              
операции с автомобилем           4279          400    0.093480       90.652022
операции с недвижимостью        10751          780    0.072551       92.744861
получение образования            3988          369    0.092528       90.747242
проведение свадьбы               2313          183    0.079118       92.088197

<div class="alert alert-warning">
<font size="5"><b>Комментарий ревьюера</b></font>

Совет: Аналогично и с категоризацией данного параметра. Проводить повторно ее не стоит. Она уже сделана ранее. 

</div>

**Вывод:** 
Операции с автомобилем:
С количеством 4279 кредитов, 400 из которых не были возвращены, процент возврата составляет 90.65%. Это довольно низкий процент, что может указывать на определенные риски при кредитовании под автомобили.

Операции с недвижимостью:
Эта категория имеет наибольшее количество кредитов (10751) и наименьший процент невозвратов с процентом возврата 92.74%. Это говорит о том, что клиенты, берущие кредиты на недвижимость, в целом более надежны.

Получение образования:
С 3988 кредитами процент возврата составляет 90.75%. Этот показатель схож с процентом возврата по кредитам на автомобили, что может свидетельствовать о рисках в данной категории.

Проведение свадьбы:
Процент возврата в этой категории составляет 92.09%, что указывает на то, что клиенты, берущие кредиты для свадебных мероприятий, также имеют высокий уровень надежности.

Клиенты, берущие кредиты на операции с недвижимостью, показывают наиболее высокий процент возврата, что делает эту категорию менее рискованной. В то время как кредиты на автомобили имеют самый низкий процент возврата.

<div class="alert alert-danger">
<font size="5"><b>Комментарий ревьюера</b></font>

Ошибка: Вывод правильный. Но попрошу все же доработать пункт. На данном шаге рекомендуется использовать метод сводных таблиц (pivot_table()). Однако он является аналогом метода groupby(). Изучи метод сводных таблиц и примени его хотя бы в одном пункте анализа.

</div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* человеческий фактор, технические сбои, несовместимости при миграции данных.

<div class="alert alert-success">
<font size="5"><b>Комментарий ревьюера</b></font>

Успех: Да, вполне возможные причины. 

</div>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Медиана является репрезентацией середины выборки то есть центр данных. В то время как срединное значение будет будет более подвержено искажениям при аномально высоких или низких значениях.

<div class="alert alert-success">
<font size="5"><b>Комментарий ревьюера</b></font>

Успех: Все верно, медиана более устойчива к наличию выбросов в данных.

</div>

### Шаг 4: общий вывод.

Напишите ваш общий вывод.

Исследование зависимости различных факторов, таких как количество детей, семейное положение, уровень дохода и цели кредита, от возврата кредита в срок предоставило ценные данные, которые могут быть полезны для кредитных учреждений и других заинтересованных сторон. Вот ключевые выводы из анализа:

Количество детей:
Анализ показал, что процент возврата кредита варьируется в зависимости от количества детей. В общем, заемщики с меньшим количеством детей показывают более высокий уровень возврата, что может свидетельствовать о более низких финансовых нагрузках и, следовательно, о большей надежности.

Семейное положение:
Результаты показали, что семейное положение влияет на возврат кредита. Заемщики, состоящие в браке или находящиеся в гражданском браке, имеют более высокий процент возврата по сравнению с теми, кто не состоит в браке. Это может указывать на более стабильные финансовые условия и большую ответственность.

Уровень дохода:
Проанализировав доходные категории, можно заметить, что заемщики с высоким доходом показывают лучший уровень возврата. Это подтверждает общепринятое мнение о том, что финансовая стабильность и доходы напрямую влияют на возможность выполнения долговых обязательств.

Цели кредита:
Разные цели кредита также влияют на возврат в срок. Например, кредиты на недвижимость демонстрируют более высокий процент возврата по сравнению с кредитами на автомобили. Это может свидетельствовать о том, что заемщики, вкладывающие средства в недвижимость, более серьезно подходят к своим финансовым обязательствам.


<div class="alert alert-success">
<font size="5"><b>Комментарий ревьюера</b></font>

Успех: Выводы описаны грамотно. Приведены ответы на главные вопросы проекта. В выводах можно приводить полученные ранее значения. Также можно расписать все, что было сделано в ходе проведения работы. Еще лучше будет, если приведешь рекомендации для компании по дальнейшим действиям. 

</div>

<div class="alert alert-info">
<font size="5"><b>Комментарий ревьюера</b></font>

Ты проделала большую работу, молодец! Ты овладела важными на данном этапе обучения инструментами, а также довольно качественно их используешь. Критических замечаний немного. Мне важно лишь убедиться в том, что метод сводных таблиц ты освоила. Также есть несколько желтых комментариев. Думаю, исправить и их тебе не составит труда. Жду твою работу :)

</div>